README.md

```diff
@@ -1,31 +1,784 @@
+
+*AlamofireRequestPerformer*
+
+`AlamofireRequestPerformer` is implementation over [Alamofire](https://github.com/Alamofire/Alamofire). By default it uses `AlamofireJsonDataEncoder` (encoder for JSON) as `DataEncoder` but this can be changed in its `init`. 
+
+`AlamofireJsonDataEncoder` knows special type of `InputEncoding` and that is `FormInputEncoding` which supports content type `application/x-www-form-urlencoded`.
+
+### Header
+
+```Swift
+protocol Header: RequestModifier {
+    
+    var name: String { get }
+    var value: String { get }
+}
+```
+
+`Header` is a special type of `RequestModifier`, which represents headers of the request. Registered instances of `Header` are added to `Request` by internal `RequestEnhancer`.
+
+#### Predefined headers
+
+`Headers` gathers all predefined headers. They are represented as nested structs added by extensions. For example this is implementation of `Accept`:
+
+```Swift
+extension Headers {
+    
+    struct Accept: Header {
+        
+        let name = "Accept"
+        
+        let value: String
+        
+        init(value: String) {
+            self.value = value
+        }
+    }
+}
+
+extension Headers.Accept {
+    
+    static let applicationJson = Headers.Accept(value: "application/json")
+    static let textPlain = Headers.Accept(value: "text/plain")
+}
+```
+
+Another headers can be added the same way (as well as concrete values of the headers).
+
+Currently there are these headers: `Accept`, `ContentType`, `Charset`.
+
+`Custom` is a special type of header. It is meant to by used if some header is needed only in one place, so it is not worth it to create a new struct.
+
+### FetcherError
+
+```Swift
+enum FetcherError: Error {
+    case requestError(Error)
+    case invalidStatusCode
+    case nilValue
+    case custom(Error)
+    case unknown
+}
+```
+
+`FetcherError` is used as the type of error in `Response.result`. 
+
+### ErrorHandler
+
+```Swift
+protocol ErrorHandler {
+    
+    func canResolveError(response: Response<SupportedType>) -> Bool
+    
+    func resolveError(response: Response<SupportedType>, callback: (Response<SupportedType>) -> Void)
+}
+```
+
+`ErrorHandler` tells `Fetcher` what to do if `Response.result` is failure. It can be set in its `init` with parameter named `errorHandler`. Default one is `NoErrorHandler`.
+
+When resolving error `Fetcher` first calls `canResolveError`. If it returns `false`, then there is no change in behavior and callback is called with unchanged `Response`. If `true`, then `resolveError` is called instead of the callback. However the callback is passed as a closure, which you can call with the modified `Response`.
+
+#### BaseStatusCodeErrorHandler
+
+`BaseStatusCodeErrorHandler` is "abstract" class for implementations that resolves errors based on the status code of the response. It has similar `init` to `StatusCodeResponseVerifier`. `canResolveError` returns `true` if the status code is one of the codes from `init`.
```

Comment:
"`BaseStatusCodeErrorHandler` is an "abstract" class ..."
maybe use "constructor" somewhere instead of init everywhere? I feel very inited from reading that many inits